### DISCUSSION TITLE
i want use copilot

### DISCUSSION THREAD

**Discussion by chenrubin1108** [community/community#10216] [Copilot Conversations] [answered]
this is my information
```
vscode version 1.63.2
os: windows_NT x64  10.0.19042
copilot  v1.7.4608
```
I want to log into Github for authorization copilot.ant then vscode show me this
```
[ERROR] [default] [2022-01-17T06:47:23.509Z] GitHub Copilot could not connect to server. Extension activation failed: "request to https://vscode-auth.github.com/token?code=xxxxxxxx&state=xxxxxxxx failed, reason: Client network socket disconnected before secure TLS connection was established"
```
i must be use proxy.  can not close Proxy Support.  if i dont use proxy  i do not login into github.so what can i do.I want to use Copilot  it is amazing!    I wish someone could help me thanks!

**Comment by davecheney** [comment, marked as answer]
I'm sorry that you are unable to use copilot at the moment. We know that supporting SSL intercepting proxies is a requirement for some of our users. Sadly I cannot share a timeline when we plan to implement this.

**Reply by OctoDat** [reply]
Has there been any updates?

**Comment by xianshenglu** [comment]
Months before, I can use it in my company. However, not sure from when, I got `GitHub Copilot could not connect to server. Extension activation failed: "Client network socket disconnected before secure TLS connection was established"`. Then I can not use copilot any more. Any ideas?

**Reply by xianshenglu** [reply]
@davecheney

**Reply by davecheney** [reply]
It looks like something is intercepting your SSL traffic. Do you this could be the case?

**Reply by xianshenglu** [reply]
> It looks like something is intercepting your SSL traffic. Do you this could be the case?

Under company's proxy. So, what can I do?

**Reply by davecheney** [reply]
I'm afraid Copilot doesn't work with proxies or SSL intercepting firewalls at the moment. I know its something we're looking into, but I cannot give you an estimate this work might be complete.

**Comment by p333ter** [comment]
Any news @davecheney ?

**Comment by TheKevinWang** [comment]
still having this issue. not sure why they didnt think about this from the beginning since most organizations have ssl interception or why this is still a problem half a year later

**Reply by DaniellePashayan** [reply]
second this

**Reply by OctoDat** [reply]
Any updates?
